packages/performance/src/services/transport_service.ts

```diff
@@ -73,60 +82,134 @@ function processQueue(timeOffset: number): void {
+      }
+      return res.json();
     })
-      .then(res => {
-        if (!res.ok) {
-          consoleLogger.info('Call to Firebase backend failed.');
+    .then(res => {
+      // Find the next call wait time from the response.
+      const transportWait = Number(res.nextRequestWaitMillis);
+      let requestOffset = DEFAULT_SEND_INTERVAL_MS;
+      if (!isNaN(transportWait)) {
+        requestOffset = Math.max(transportWait, requestOffset);
+      }
+
+      // Delete request if response include RESPONSE_ACTION_UNKNOWN or DELETE_REQUEST action.
+      // Otherwise, retry request using normal scheduling if response include RETRY_REQUEST_LATER.
+      const logResponseDetails: LogResponseDetails[] = res.logResponseDetails;
+      if (logResponseDetails && logResponseDetails.length > 0) {
+        if (logResponseDetails[0].responseAction === 'RETRY_REQUEST_LATER') {
+          queue = [...staged, ...queue];
+          consoleLogger.info(`Retry transport request later.`);
         }
-        return res.json();
-      })
-      .then(res => {
-        const wait = Number(res.next_request_wait_millis);
-
-        // Find the next call wait time from the response.
-        const requestOffset = isNaN(wait)
-          ? DEFAULT_SEND_INTERVAL_MS
-          : Math.max(DEFAULT_SEND_INTERVAL_MS, wait);
-        remainingTries = DEFAULT_REMAINING_TRIES;
-        // Schedule the next process.
-        processQueue(requestOffset);
-      })
-      .catch(() => {
-        /**
-         * If the request fails for some reason, add the events that were attempted
-         * back to the primary queue to retry later.
-         */
-        queue = [...staged, ...queue];
-        remainingTries--;
-        consoleLogger.info(`Tries left: ${remainingTries}.`);
-        processQueue(DEFAULT_SEND_INTERVAL_MS);
-      });
-  }, timeOffset);
+      }
+
+      remainingTries = DEFAULT_REMAINING_TRIES;
+      // Schedule the next process.
+      processQueue(requestOffset);
+    });
+}
+
+function sendEventsToCc(data: TransportBatchLogFormat): Promise<void> {
+  return fetch(SettingsService.getInstance().logEndPointUrl, {
+    method: 'POST',
+    body: JSON.stringify(data)
+  })
+    .then(res => {
+      if (!res.ok) {
+        consoleLogger.info('Call to Firebase backend failed.');
+      }
+      return res.json();
+    })
+    .then(res => {
+      const wait = Number(res.next_request_wait_millis);
+      // Find the next call wait time from the response.
+      const requestOffset = isNaN(wait)
+        ? DEFAULT_SEND_INTERVAL_MS
+        : Math.max(DEFAULT_SEND_INTERVAL_MS, wait);
+      remainingTries = DEFAULT_REMAINING_TRIES;
+      // Schedule the next process.
+      processQueue(requestOffset);
+    });
+}
+
+function postToTransportEndpoint(
+  transportUrl: string,
+  transportKey: string,
+  data: TransportBatchLogFormat
+): Promise<Response> {
+  const transportFullUrl = transportUrl.concat('?key=', transportKey);
```

Comment:
We would probably better off having a method to get the url in settings_service.